### PR TITLE
add dockerhub login before version check

### PR DESCRIPTION
**Pull request by zhonghui12**
Signed-off-by: Zhonghui Hu <zh0512xx@gmail.com>

*Description of changes:*
Add dockerhub login before version check.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by zhonghui12** [comment]
Don't merge the code until our pipeline is redeployed.

File: scripts/publish.sh [hunk 1]
```diff
@@ -351,6 +351,19 @@ verify_ecr() {
...
+
+	# Logout when the script exits
+	trap cleanup EXIT
+	cleanup() {
+		docker logout
+	}
+
+	# login to DockerHub
+	docker login -u "${username}" --password "${password}"
+
```
Comment (hossain-rayhan) [inline scripts/publish.sh:366]: We are doing this in at least two different places. Is it possible to extract them in a function to avoid duplicate code?

**Review by hossain-rayhan — APPROVED** [review]

File: scripts/publish.sh (hunk 1 above)
Comment (zhonghui12) [reply scripts/publish.sh:366]: good point. We can do that to make the code more concise

Comment (PettitWesley) [reply scripts/publish.sh:366]: +1

**Review by PettitWesley — APPROVED** [review]

**Review by PettitWesley — APPROVED** [review]

**Pull request merged**